src/Transformer/FractalTransformer.php

```diff
@@ -76,7 +76,9 @@ public function transform($response, $transformer, Binding $binding, Request $re
         }
 
         if ($response instanceof EloquentCollection) {
-            $response->load($this->fractal->getRequestedIncludes());
+            $requestedIncludes = $this->fractal->getRequestedIncludes();
+            $eagerLoads = $this->getEagerLoads($transformer, $requestedIncludes);
+            $response->load($eagerLoads);
```

Comment:
I got interested in true eager loading because in my API I'm pulling ids from related objects that may or may not be included, according to JSON-API spec.

In my `UserTransformer->transform()`:

```php
    if(count($user->posts)) {
        $links['posts'] = $user->posts->lists('id');
    }
```

This code is always called, so I get `n + 1` on the user index if I don't request the inclusion of posts.